src/12_output_CHSA_DA_lookup.R

```diff
@@ -0,0 +1,541 @@
+# each row/year is expanded to 2019:2023
+gcs_chsa_db_lookup_2021 <- gcs_data %>%
+  count(
+    CHSA,
+    DAUID = DA_2021,
+    DBUID = DB_2021,
+    name = "CNT_POSTALCODE"
+  ) %>%
+  tidyr::expand_grid(YEAR = 2019:2023) %>%
+  left_join(chsa_data)
+
+
+# start working on DB2021 and CHSA
+gcs_chsa_db_lookup_2016 <- gcs_data %>%
+  count(
+    CHSA,
+    DAUID = DA_2016,
+    DBUID = DB_2016,
+    name = "CNT_POSTALCODE"
+  ) %>%
+  tidyr::expand_grid(YEAR = 2016:2018) %>%
+  left_join(chsa_data)
+
+
+gcs_chsa_db_lookup_2016_2023 <- gcs_chsa_db_lookup_2021 %>%
+  bind_rows(gcs_chsa_db_lookup_2016) |>
+  select(-CNT_POSTALCODE) |>
+  select(YEAR, CHSA, CHSA_NAME, DBUID)
+
+gcs_chsa_db_lookup_2016_2023 |> glimpse()
+# 234,073
+#########################################################################
+# Use Jonathan's TMF file, limitation is that it only has DB 2021 and DA 2021.
+# So may take both and make a set of CHSA, DB, and YEAR
+#########################################################################
+
+#Setting pathsAdd commentMore actions
+bc_stats_project_lan <- config$file_path$bc_stats_project
+
+Path_conversion <- file.path(
+  bc_stats_project_lan,
+  "ConversionTables",
+  "TranslationMasterFiles",
+  "2025"
+)
+
+
+#Loading the conversion table
+
+#Conversion table
+
+conversion_table <- readxl::read_xls(file.path(
+  Path_conversion,
+  paste0(
+    "Translation_Master_File_",
+    "2021_Census_Completed_",
+    "Provincial_Runs_2025_01_10",
+    " (identifiers).xls"
+  )
+))
+
+conversion_table |>
+  # glimpse()
+  count(DISSEMINATION_AREA_ID)
+# 7,848
+
+db_da_chsa_2021_jonathan <- conversion_table |>
+  select(
+    CHSA,
+    DBUID = "DISSEMINATION_BLOCK_ID",
+    DAUID = "DISSEMINATION_AREA_ID"
+  ) |>
+  distinct() |>
+  mutate(CHSA = as.character(CHSA)) |>
+  left_join(chsa_data)
+
+db_da_chsa_2021_jonathan |> glimpse()
+# 52423 DBs
+
+db_da_chsa_2016_2023_jonathan <- db_da_chsa_2021_jonathan |>
+  expand_grid(YEAR = 2016:2023) |>
+  select(YEAR, CHSA, CHSA_NAME, DBUID)
+
+db_da_chsa_2016_2023_jonathan |> glimpse()
+# 419,384 rows
+
+db_da_chsa_lookup_2016_2023 <- gcs_chsa_db_lookup_2016_2023 |>
+  bind_rows(db_da_chsa_2016_2023_jonathan) |>
+  distinct()
+
+db_da_chsa_lookup_2016_2023 |>
+  glimpse()
+# 429,323 rows
+# So we recover some (10000) DB2016 from GCS table
+
+#################################################################
+# DB population from Jonathan
+#################################################################
+#Setting paths
+bc_stats_project_lan <- config$file_path$bc_stats_project
+
+# replace this data source with Jonathan's DB population estimate from 2016 to 2023
+
+# file path of DB population estimates
+db_population_file_path = file.path(
+  bc_stats_project_lan,
+  config$file_path$population_folder
+)
+
+# list all the rds files
+db_population_files = list.files(db_population_file_path, full.names = T)
+# read all the rds files
+db_population_df = db_population_files %>%
```

Comment:
When I read this in, it comes in with a column "Population", not "Population_2021_Adjusted", so the next line (233) fails.